### PR TITLE
Disallow `BLK<SIMD>`

### PR DESCRIPTION
**Pull request by SingleAccretion**
For a very long time, both `OBJ/BLK<SIMD>` and `IND<SIMD>` representations for a SIMD-typed load have been legal. There is no point in having both, so this change disallows the former (as a less canonical one).

Plus a number of supporting changes (see individual commits).

[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=241041&view=ms.vss-build-web.run-extensions-tab).

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

File: src/coreclr/jit/compiler.h [hunk 1]
```diff
@@ -2833,6 +2831,19 @@ class Compiler
 
     GenTreeIndir* gtNewIndir(var_types typ, GenTree* addr, GenTreeFlags indirFlags = GTF_EMPTY);
 
+    GenTree* gtNewLoadValueNode(
```
Comment (tannergooding) [inline src/coreclr/jit/compiler.h:2834]: Do you plan on a similar `gtNewStoreValueNode` for parity?

Right now that's one of a few nodes, including `ASG` iirc?

**Review by tannergooding — APPROVED** [review]

File: src/coreclr/jit/compiler.h (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/jit/compiler.h:2834]: > Do you plan on a similar `gtNewStoreValueNode` for parity?

Yep, these are all part of the `ASG` removal project.

File: src/coreclr/jit/morph.cpp [hunk 2]
```diff
@@ -9863,7 +9863,7 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
             // Only do this optimization when we are in the global optimizer. Doing this after value numbering
             // could result in an invalid value number for the newly generated GT_IND node.
             // We skip INDs with GTF_DONT_CSE which is set if the IND is a location.
-            if (op1->OperIs(GT_COMMA) && fgGlobalMorph && ((tree->gtFlags & GTF_DONT_CSE) == 0))
+            if (!varTypeIsSIMD(tree) && op1->OperIs(GT_COMMA) && fgGlobalMorph && ((tree->gtFlags & GTF_DONT_CSE) == 0))
```
Comment (jakobbotsch) [inline src/coreclr/jit/morph.cpp:9866]: Is this for CQ purposes?

Comment (jakobbotsch) [reply src/coreclr/jit/morph.cpp:9866]: Gotcha. It would be nice to remove this entirely at some point.

**Comment by jakobbotsch** [comment]
I *think* the failures are current known ones.

**Comment by SingleAccretion** [comment]
I looked as well and believe so to. [Jitstress run](https://dev.azure.com/dnceng-public/public/_build/results?buildId=244218&view=ms.vss-test-web.build-test-results-tab).

**Pull request merged**